LMS/User/Http/Requests/UpdateMeRequest.php

```diff
@@ -21,6 +21,7 @@ public function rules()
             'name' => 'string|min:6|max:80',
             'phone_number' => 'required|string',
             'birthdate' => 'required|date',
+            'username' => 'required|string',
```

Comment:
No caso do campo username, eu fico na dúvida de como proceder com ele. Ele tem que ser único no banco e só ser validado após o formulário passar pelo primeiro estágio de validação.

FormRequest -> Repository com validação de username

... pelo fato de ser algo que pode não ter sido mudado anteriormente no form e você mandar mesmo assim.

Reply:
Cara, vou admitir um lance pra vc KKKKKKKKKKKK
Eu tinha colocado 
```
'username' => 'required|string|unique:users,username'
``` 

Fui testar e deu erro que o username já existia KKKKKKKKKKKK ai pra meter o loco do jeito junior que sou, só tirei esse unique ahahaha

Tem alguma ideia de como eu possa aplicar essa correção ? Poderia me indicar algum link para que eu possa estudar e aplicar aqui no nosso projetinho ?

Reply:
@NicolasPereira você pode estudar sobre as Custom Rules do Laravel. Você pode criar o seu próprio fluxo de validação.
https://laravel.com/docs/8.x/validation#custom-validation-rules

Reply:
Você pode passar um terceiro parâmetro pra regra unique contendo o id do usuário atual, fazendo com que ele ignore esse registro no banco e busque apenas por outras ocorrências.

e.g.
```php
'username' => 'required|string|unique:users,username,' . Auth::id()`
```

Forcing A Unique Rule To Ignore A Given ID
https://laravel.com/docs/8.x/validation#rule-unique